src/Sylius/Bundle/ApiBundle/Validator/Constraints/CorrectOrderAddressValidator.php

```diff
@@ -0,0 +1,61 @@
+<?php
+
+/*
+ * This file is part of the Sylius package.
+ *
+ * (c) Paweł Jędrzejewski
+ *
+ * For the full copyright and license information, please view the LICENSE
+ * file that was distributed with this source code.
+ */
+
+declare(strict_types=1);
+
+namespace Sylius\Bundle\ApiBundle\Validator\Constraints;
+
+use Sylius\Bundle\ApiBundle\Command\Checkout\AddressOrder;
+use Sylius\Component\Core\Model\AddressInterface;
+use Symfony\Component\Intl\Countries;
+use Symfony\Component\Validator\Constraint;
+use Symfony\Component\Validator\ConstraintValidator;
+use Webmozart\Assert\Assert;
+
+final class CorrectOrderAddressValidator extends ConstraintValidator
+{
+    public function validate($value, Constraint $constraint)
+    {
+        /** @var AddressOrder $value */
+        Assert::isInstanceOf($value, AddressOrder::class);
+
+        /** @var CorrectOrderAddress $constraint */
+        Assert::isInstanceOf($constraint, CorrectOrderAddress::class);
+
+        $this->validateAddress($value->billingAddress, $constraint);
+        $this->validateAddress($value->shippingAddress, $constraint);
+    }
+
+    private function validateAddress(?AddressInterface $address, CorrectOrderAddress $constraint): void
+    {
+        if ($address === null) {
+            return;
+        }
+
+        /** @var string|null $countryCode */
+        $countryCode = $address->getCountryCode();
+
+        if ($countryCode === null) {
+            $this->context->addViolation(
+                $constraint->addressWithoutCountryCodeCanNotExistMessage
+            );
+
+            return;
+        }
+
+        if (!Countries::exists($countryCode)) {
```

Comment:
I believe we should use the `countryRepository` and check if the country exists within Sylius